core/domain/state_domain.py

```diff
@@ -174,6 +175,208 @@ def validate(self):
         self.hint_content.validate()
 
 
+class Image(object):
+    """Value object representing the image for the state."""
+    def __init__(self, src, placeholder, instructions):
+        """Constructs an Image domain object.
+
+        Args:
+            src: str. The source of an image.
+            placeholder: bool. Whether the image is a placeholder.
+            instructions: str. Instructions for the image which needs in place
+                of a placeholder.
+        """
+        self.src = src
+        self.placeholder = placeholder
+        self.instructions = instructions
+        self.validate()
+
+    def to_dict(self):
+        """Returns a dict representing this image domain object.
+
+        Returns:
+            dict. A dict, representation of the image instance.
+        """
+        return {
+            'src': self.src,
+            'placeholder': self.placeholder,
+            'instructions': self.instructions
+        }
+
+    @classmethod
+    def from_dict(cls, image_dict):
+        """Return an Image object from a dict.
+
+        Args:
+            image_dict: dict. The dict representation of image object.
+
+        Returns:
+            Image. The corresponding Image domain object.
+        """
+        src = image_dict['src']
+        placeholder = image_dict['placeholder']
+        instructions = image_dict['instructions']
+
+        return cls(src, placeholder, instructions)
+
+    def validate(self):
+        """Validates all properties of an Image object.
+
+        Raises:
+            ValidationError: One or more attributes of the Image are not valid.
+        """
+        filename_re = r'^[A-Za-z0-9+/_-]*\.((png)|(jpeg)|(gif)|(jpg))$'
+        if not re.match(filename_re, self.src):
+            raise utils.ValidationError(
+                'Invalid filepath, received %s' %
+                self.src)
+        if not isinstance(self.placeholder, bool):
+            raise utils.ValidationError(
+                'Expected placeholder to be bool, received %s' %
+                self.placeholder)
+        if not isinstance(self.instructions, basestring):
```

Comment:
hmmm... I think we should make these instructions mandatory for the placeholders. I see for Images we have to provide alt and caption, otherwise one cannot use the image for the exploration. (not sure whether for both alt and captionor one)

/cc @nithusha21